### PR TITLE
change(merge): Require 2 reviews for PRs with an extra-reviews label

### LINKED ISSUE #5996: Require two reviewers for complex or risky code changes
## Motivation

We'd like to require two reviewers for complex or risky code changes.

We could do that by adding a new Mergify workflow at the top of the workflow list for these labels:
- A-compatibility
- A-concurrency
- A-consensus
- I-consensus
- C-security
- I-crash
- I-destructive
- I-hang
- I-privacy
- I-unbounded-growth
- I-unsound

We'd also like two reviewers for unsafe code, we could:
1. auto-label PRs that change the paths for unsafe reviewers with a new A-unsafe label
2. Add that label to the Mergify list

### Testing

We'll see if it works for the first consensus or compatibility PR.

### PR DESCRIPTION
**Pull request by teor2345**
## Motivation

This PR makes sure we do two reviews on PRs with the `extra-reviews` label.

Closes #5996

### Specifications

https://docs.mergify.com/examples/#using-labels-to-enable-disable-merge

## Solution

- Require 2 reviews if the `extra-reviews` label is on a PR
- Automatically add an `extra-reviews` label with release drafter to PRs with titles with:
  - remov*
  - deprecat*
  - crash*
  - destr*
  - unsafe*
- Clean up the auto-labelling in release-drafter.yml

## Review

This would be good to have for PR #7053.

### Reviewer Checklist

  - [ ] Are the PR labels correct?
  - [ ] Does the code do what the ticket and PR says?
    - [ ] Does it change concurrent code, unsafe code, or consensus rules?
  - [ ] How do you know it works? Does it have tests?

### CODE REVIEW

File: .github/release-drafter.yml [hunk 1]
```diff
@@ -164,9 +158,9 @@ template: |
   ### Breaking Changes
   
   This release has the following breaking changes:
-  - *TODO*: Check the `Removed` section for any breaking changes
+  - *TODO*: Check the `Removed` and  `Deprecated` sections for any breaking changes
```
Comment (teor2345) [inline .github/release-drafter.yml]:
```suggestion
  - *TODO*: Check the `Removed` and `Deprecated` sections for any breaking changes
```

**Comment by teor2345** [comment]
@mergifyio update

**Review by gustavovalverde — APPROVED** [review]
LGTM! And most of us agreed using this approach and label wording

**Comment by teor2345** [comment]
This PR contains a mergify config, so it needs a manual merge.

**Pull request merged**